zip-0173.rst

```diff
@@ -0,0 +1,469 @@
+`here <https://github.com/sipa/ezbase32/>`_.
+
+Properties
+''''''''''
+
+The following table summarizes the chances for detection failure (as multiples of
+1 in 10\ :sup:`9`).
+
++-------------------------------------+-----------------------------------------------+
+|            Window length            |          Number of wrong characters           |
++--------+----------------------------+-------+-------+-------+-------+-------+-------+
+| Length | Description                |  ≤4   |   5   |   6   |   7   |   8   |  ≥9   |
++--------+----------------------------+-------+-------+-------+-------+-------+-------+
+|    8   | Longest detecting 6 errors |   0                   | 1.127 | 0.909 |  n/a  |
++--------+----------------------------+---------------+-------+-------+-------+-------+
+|   18   | Longest detecting 5 errors |   0           | 0.965 | 0.929 | 0.932 | 0.931 |
++--------+----------------------------+-------+-------+-------+-------+-------+-------+
+|   19   | Worst case for 6 errors    |   0   | 0.093 | 0.972 | 0.928 | 0.931         |
++--------+----------------------------+-------+-------+-------+-------+---------------+
+|   39   | Length ≤ 39 characters     |   0   | 0.756 | 0.935 | 0.932 | 0.931         |
++--------+----------------------------+-------+-------+-------+-------+---------------+
+|   59   | Length ≤ 59 characters     |   0   | 0.805 | 0.933 | 0.931                 |
++--------+----------------------------+-------+-------+-------+-----------------------+
+|   71   | Length ≤ 71 characters     |   0   | 0.830 | 0.934 | 0.931                 |
++--------+----------------------------+-------+-------+-------+-----------------------+
+|   89   | Longest detecting 4 errors |   0   | 0.867 | 0.933 | 0.931                 |
++--------+----------------------------+-------+-------+-------+-----------------------+
+
+TODO: fill in table for a Sapling payment address, and check the following paragraph.
+
+This means that when 5 changed characters occur randomly distributed in the 77
+characters (excluding the separator) of a Sapling payment address, there is a chance
+of at most 0.867 per billion that it will go undetected. When those 5 changes occur
+randomly within a 19-character window, that chance goes down to 0.093 per billion.
+As the number of errors goes up, the chance converges towards 1 in 2\ :sup:`30` =
+0.931 per billion.
+
+The chosen code performs reasonably well up to 1023 characters, but is best for
+lengths up to 89 characters (excluding the separator). Since the search for suitable
+codes was based on the requirements for Bitcoin P2WPKH and P2WSH addresses, it is
+not quite optimal for the address lengths used by Sapling, but the advantages of
+compatibility with existing Bitcoin libraries were considered to outweigh this
+consideration.
+
+
+Acknowledgements
+================
+
+This document is closely based on BIP 173 written by Pieter Wuille and Greg Maxwell,
```

Comment:
Why are we re-specifying Bech32? AFAIK Zcash-Bech32 is identical to Bitcoin-Bech32 except that the hard length limit is raised from 90 to 1024.

Can we not just refer to BIP 173 for the specification of Bech32, and then specify the Zcash differences and consensus-rule effects here? That seems like it would be significantly easier to both write and reason about.

Reply:
If you read BIP 173, there are quite a few things it says that are just plain wrong or inapplicable for Zcash. The ZIP fixes those.

Reply:
ACK. Le sigh. _Puts on reviewing goggles_